projects/igniteui-angular/src/lib/accordion/accordion.component.ts

```diff
@@ -231,14 +234,68 @@ export class IgxAccordionComponent implements AfterContentInit, AfterViewInit, O
         }
 
         this.panels.forEach(panel => {
-            if (!panel.collapsed) {
+            if (!panel.collapsed && !panel.header.disabled) {
                 panel.collapse();
             }
         });
     }
 
-    public handleKeydown(ev) {
+    public handleKeydown(event: KeyboardEvent, panel: IgxExpansionPanelComponent) {
+        const key = event.key.toLowerCase();
+        if (!(ACCORDION_NAVIGATION_KEYS.has(key))) {
+            return;
+        }
+        this.enabledPanels = this._panels.filter(p => !p.header.disabled);
```

Comment:
You could leave a comment here - if we ever want to improve the performance of the accordion, `enabledPanels` could also be cached (by making a `disabledChange` emitter on the panel header).
However, filtering the panels is trivial in most cases - leave it as is for the initial implementation.